mungegithub/features/repo-updates.go

```diff
@@ -174,14 +214,77 @@ func decodeAssignmentConfig(path string, config *assignmentConfig) error {
 	return parseYaml.Unmarshal([]byte(meta), &config)
 }
 
-func (o *RepoInfo) updateRepoUsers() error {
-	out, err := o.GitCommand([]string{"pull"})
+func (o *RepoInfo) updateRepoAliases() error {
+	if len(o.aliasFile) == 0 {
+		return nil
+	}
+
+	// read and check the alias-file.
+	fileContents, err := o.aliasReader.read()
+	if os.IsNotExist(err) {
+		glog.Infof("Missing alias-file (%s), using empty alias structure.", o.aliasFile)
+		o.aliasData = &aliasData{}
+		return nil
+	}
 	if err != nil {
-		glog.Errorf("Unable to run git pull:\n%s\n%v", string(out), err)
-		return err
+		return fmt.Errorf("Unable to read alias file: %v", err)
+	}
+
+	var data aliasData
+	if err := parseYaml.Unmarshal(fileContents, &data); err != nil {
+		return fmt.Errorf("Failed to decode the alias file: %v", err)
+	}
+
+	o.aliasData = &data
+	o.aliasData = normalizeAliases(data)
+	return nil
+}
+
+// normalizeAliases normalizes all entries in the alias data
+// so that we can do case-insensitive resolution of aliases
+func normalizeAliases(rawData aliasData) *aliasData {
+	normalizedData := aliasData{AliasMap: map[string][]string{}}
+	for alias, users := range rawData.AliasMap {
+		normalizedData.AliasMap[normalizeUser(alias)] = normalizeUsers(users)
+	}
+	return &normalizedData
+}
+
+// expandAllAliases expands all of the aliases in the
+// lists of approvers and reviewers for paths we track
+func (o *RepoInfo) expandAllAliases() {
+	for filePath, approvers := range o.approvers {
+		o.resolveAliases(&approvers)
+		o.approvers[filePath] = approvers
+	}
+
+	for filePath, reviewers := range o.reviewers {
+		o.resolveAliases(&reviewers)
+		o.reviewers[filePath] = reviewers
```

Comment:
since sets are reference types (`map[string]struct{}` in this case) the range statement's copy still refers to the same underlying map.  See [this example](https://play.golang.org/p/IN_yXs4fzT).